### PR TITLE
feat: Add option to disable automatic loading of JavaScript

### PR DESCRIPTION
**Pull request by svenopdehipt**
Add the option to disable automatic loading of JavaScript, which is necessary to conform to new EU-laws which don't permit to load from third party domains per the user agrees to (which he can't when it's happening on page load).

### CODE REVIEW

**Comment by navaronbracke** [comment]
`which is necessary to conform to new EU-laws` what specific law are you referring to here?

**Comment by svenopdehipt** [comment]
https://blog.runcloud.io/google-fonts-gdpr/ This is an example with google fonts, but it also applies to other content which is loaded from 3rd-party servers. The law itself isn't new but the verdict is.

**Comment by navaronbracke** [comment]
In that case I'd switch to using a locally bundled version of the library? (referring to the locally hosting of the asset)

**Comment by svenopdehipt** [comment]
Yes, you have to host the library yourself

**Comment by navaronbracke** [comment]
What I actually meant is that instead of disabling the automatic loading of the JS library, we change the loading of the library as follows:

- put the library JS sources under the `/web` folder in the plugin (like how other plugins do it when targeting the web)
- load the library from the asset in `/web`
- keep the behavior of automatically loading the library (now that it loads from the asset)

This way we are compliant with the law, and the consumers of the plugin don't have to do anything extra.

**Comment by svenopdehipt** [comment]
I didn't know of this possibility. This would also solve the problem.

**Comment by navaronbracke** [comment]
@svenopdehipt I had a quick look at it at it seems that my assumption was wrong.

If I add `./web/zxing.js` to the plugin and then try to load that, it does not work.
I get 

```
Refused to execute script from 'http://localhost:63940/zxing.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.
```
due to the fact that the file is not found. It does not show up in the file system in Chrome devtools either. Probably because DDC/Dart2js does not include it in the build of the web app.

Also when looking at first party plugins (i.e. `google_maps_flutter`) they do include the script manually in their index.html

What if we move the loading of the library into an `initialize()` method? This method then would load the script in a script tag like how it happens now. Just on demand, instead of automatically.

I did the same for a different plugin for which I am adding web support.
In that scenario, a developer can implement asking permission before loading the library.

WDYT?

**Comment by svenopdehipt** [comment]
I would have time to change the code today. The only question if the function shouldn't be placed in src/mobile_scanner_controller.dart instead of mobile_scanner_web.dart. When importing mobile_scanner_web.dart, which imports dart:html, I have to do a conditional import with a stub as a fallback to support dart vm.

**Comment by navaronbracke** [comment]
Well, that is up to you to decide I think.

This package does not use the `plugin_platform_interface` package, the way it is supposed to.
That would eliminate the stub / conditional import. 

See also: #401

File: lib/src/mobile_scanner_controller.dart [hunk 1]
```diff
@@ -68,6 +68,13 @@ class MobileScannerController {
   static const EventChannel _eventChannel =
       EventChannel('dev.steenbakker.mobile_scanner/scanner/event');
 
+  /// Call this method to automatically load the javascript libraries, which are necessary for the plugin.
```
Comment (navaronbracke) [inline lib/src/mobile_scanner_controller.dart:71]: Same here, `JavaScript` does not apply when running on mobile / desktop.

**Pull request closed without merging**